Cryptolib/DecisionalDiffieHellman.lean

```diff
@@ -0,0 +1,52 @@
+/-
+Copyright (c) 2021 Joey Lupo. All rights reserved.
```

Comment:
The phrase  "All rights reserved" [has no legal effect](https://en.wikipedia.org/wiki/All_rights_reserved) and its meaning conflicts somewhat with open-source licensing.

Reply:
I duplicated this wording from the headers used in Mathlib. If we want to avoid using it here, that's fine with me (neither the wording with or without matches what the Apache 2.0 license says you should use, but in practice everyone understands how these abbreviations work).